mosviz/viewers/mos_viewer.py

```diff
@@ -312,9 +313,49 @@ def load_selection(self, row):
             A `row` object representing a row in the MOS catalog. Each key
             should be a column name.
         """
-        spec1d_data = nirspec_spectrum1d_reader(row['spectrum1d'])
-        spec2d_data = nirspec_spectrum2d_reader(row['spectrum2d'])
-        image_data = acs_cutout_image_reader(row['cutout'])
+
+        if "loaders" in self.catalog.meta:
+            # if loader is specified
+            if "spec1d" in self.catalog.meta["loaders"]:
+                # check .glue/config.py for custom loaders
+                if hasattr(config, self.catalog.meta["loaders"]["spec1d"]):
```

Comment:
Instead of using the `hasattr` and `getattr` builtin functions, I think it'd be more explicit and much more robust to just ask the `DataFactoryRegistry` if the data loader exists. The user may not always be adding loaders through their `config.py` file.

You're already importing the `config` module, so checking to see if the data loader exists should be as simple as

```python
spectrum1d_loader = next((x for x in config.data_factory.members if x.label ==  self.catalog.meta["loaders"]["spec1d"]),
                     next(x for x in config.data_factory.members if x.label == "default_spec1d_loader_label"))
```